src/components/layout/utils/tree-html-data-utils.ts

```diff
@@ -1,8 +1,187 @@
+import { TreeObject, HtmlComponentType } from "../../../types";
+
 /**
- * Parse HTML string to HTML
+ * Update item within tree structure
  *
- * @param html
+ * @param treeData list of TreeObject
+ * @param updatedComponent updated TreeObject
+ * @param destinationPath path of the element to be updated within tree
+ * @returns updatedTree list of TreeObjects
  */
-export const parseToHtml = (htmlString: string) => {
-  return new DOMParser().parseFromString(htmlString, "text/html").body;
+export const updateHtmlComponent = (treeData: TreeObject[], updatedComponent: TreeObject, destinationPath: string): TreeObject[] => {
+  const updatedTree: TreeObject[] = [treeData[0]];
+  if (treeData[0].id === destinationPath) {
+    updatedTree[0] = updatedComponent;
+  } else {
+    updatedTree[0].children = updateInTree(updatedTree[0].children, destinationPath, treeData[0].id, updatedComponent);
+  }
+
+  return updatedTree;
+};
+
+/**
+ * Recursive function that checks the updatedComponent children and to find item to be updated.
+ *
+ * @param treeData list of TreeObject
+ * @param destinationPath path to the item to be updated within tree
+ * @param currentPath current path inside the recursion
+ * @param updatedComponent updated TreeObject
+ * @returns list of TreeObjects
+ */
+const updateInTree = (treeData: TreeObject[], destinationPath: string, currentPath: string, updatedComponent: TreeObject): TreeObject[] => {
+  const cleanNodes: TreeObject[] = [];
+  let found = false;
+  for (const element of treeData) {
+    const node = element;
+    const fullPath = `${currentPath}/${node.id}`;
+    if (fullPath !== destinationPath) {
+      cleanNodes.push(node);
+    } else {
+      cleanNodes.push(updatedComponent);
+      found = true
+    }
+  }
+
+  if (found) {
+    return cleanNodes;
+  } else {
+    for (const element of treeData) {
+      const child = element;
+      const updatedPath = `${currentPath}/${child.id}`;
+      child.children = updateInTree(child.children ?? [], destinationPath, updatedPath, updatedComponent);
+    }
+  }
+
+  return treeData;
+};
+
+/**
+ * Convert tree object to html element
+ *
+ * @param treeObject
+ * @returns HTMLElement
+ */
+export const treeObjectToHtmlElement = (treeObject: TreeObject): HTMLElement => {
+  const element = treeObject.element;
+  element.replaceChildren();
+  if (treeObject.children) {
+    for (let i = 0; i < treeObject.children.length; i++) {
+      element.appendChild(treeObjectToHtmlElement(treeObject.children[i]));
+    }
+  }
+
+  return element;
+};
+
+/**
+ * Adds new HTML component to tree
+ * 
+ * @param treeData tree data
+ * @param newComponent new component to be added
+ * @param siblingPath path of the sibling component
+ * @returns updated tree data
+ */
+export const addNewHtmlComponent = (treeData: TreeObject[], newComponent: TreeObject, siblingPath: string) => {
+  const updatedTree: TreeObject[] = [treeData[0]];
```

Comment:
could this not just be treeData, rather than [treeData[0]]

Reply:
It needs to be casted as array in order to work